libnest/src/package/npf.rs

```diff
@@ -0,0 +1,60 @@
+use std::fs::{self, File};
+use std::marker::PhantomData;
+use std::path::{Path, PathBuf};
+
+use failure::Error;
+use tar::Archive;
+
+/// Structure representing a handle over a file contained in an NPF
+#[derive(Debug)]
+pub struct NPFFile<'explorer> {
+    file: File,
+    phantom: PhantomData<&'explorer NPFExplorer>,
+}
+
+impl<'explorer> NPFFile<'explorer> {
+    pub(crate) fn from(file: File, phantom: PhantomData<&'explorer NPFExplorer>) -> Self {
+        Self { file, phantom }
+    }
+
+    /// Retrieves the file associated with this handle
+    pub fn file(&self) -> &File {
+        &self.file
+    }
+}
+
+/// Structure representing an NPF to allow interacting with it
+#[derive(Debug)]
+pub struct NPFExplorer {
+    path: PathBuf,
+}
+
+impl NPFExplorer {
+    /// Create a NPFExplorer from a name and a path to an NPF archive
+    pub fn from(name: &str, npf_path: &Path) -> Result<Self, Error> {
+        let path = std::env::temp_dir().join(name);
```

Comment:
I think this system is weak: it prevents two NPF of the same name to be explored at the same time.

Especially, `name` seems to only be the name of the package, which will collide for two packages of the same category or two different versions of the same package.

For nest-server, i have generated a complete random name when writing a similar code. I think it's a more powerful approach to the problem, but you might have another idea?

Reply:
I totally agree on this point, I plan on fixing that (later, as the goal of this PR is to add basic support of NPFs) by "randomly" generating a directory in which to extract.